client/src/App.tsx

```diff
@@ -20,12 +21,7 @@ function App(): JSX.Element {
               <Switch>
                 <Route exact path="/login" component={Login} />
                 <Route exact path="/signup" component={Signup} />
-                <Route exact path="/dashboard">
-                  <Dashboard />
-                </Route>
-                <Route path="*">
-                  <Redirect to="/login" />
-                </Route>
+                <PrivateRoute path="/dashboard" component={Dashboard} />
```

Comment:
Just a suggestion, wouldn't it be better to use the exact param to disable partial matching for a route? :)